presto-main/src/main/java/com/facebook/presto/operator/aggregation/TypedSet.java

```diff
@@ -136,7 +139,7 @@ private void addNewElement(int hashPosition, Block block, int position)
     {
         elementType.appendTo(block, position, elementBlock);
         if (elementBlock.getSizeInBytes() > FOUR_MEGABYTES) {
-            throw exceededLocalLimit(new DataSize(4, MEGABYTE));
+            throw new PrestoException(EXCEEDED_FUNCTION_MEMORY_LIMIT, String.format("the argument of %s exceeds of %s", functionName, MAX_FUNCTION_MEMORY));
```

Comment:
Error messages should start with uppercase.

Let's make the mesasge:
```
The input to %s is too large. More than %s of memory is needed to hold the intermediate hash set.
```

Examples:
```
The input to array_distinct is too large. More than 4MB of memory is needed to hold the intermediate hash set.
The input to map-to-map cast is too large. More than 4MB of memory is needed to hold the intermediate hash set.
```

```
"array_distinct"
"map-to-map cast"
```

I changed "argument" to "input" to be more generic and it sounds a little better.